### ISSUE TITLE
Stopped bridge still generating log entries

### ISSUE THREAD

**Issue by Calorion** [hoobs-org/HOOBS#1511]
**Description**

I have a bridge with a plug-in (Hue) that is still generating log entries even though stopped. I wouldn’t think that stopped bridges would do…anything.

**Version**
4.0.100

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ ] Yes
- [x] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [x] HOOBS Box
- [ ] HOOBS SD Card
- [ ] Custom Install

**List your plugins**
Hue
Ecobee
Dummy
Ifttt
Platform Insteon
Philips Hue Sync Box

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
Sorry, I don't see how to to this. Help?
```

**Post your log**
You can get the log from the interface. This is in the Log section.

```

9/5/2021, 10:47:14 PM Hue Bridge Hue Philips hue: request 18: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:47:14 PM Hue Bridge Hue [ WARNING ] Philips hue: request 18: api error 101: link button not pressed
9/5/2021, 10:47:14 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s
9/5/2021, 10:47:29 PM Hue Bridge Hue Philips hue: request 19: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:47:29 PM Hue Bridge Hue [ WARNING ] Philips hue: request 19: api error 101: link button not pressed
9/5/2021, 10:47:29 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s
9/5/2021, 10:47:38 PM ecobee Bridge Ecobee 3 Sensors Update sensors
9/5/2021, 10:47:38 PM ecobee Bridge Ecobee 3 Sensors [Upstairs] 22.4°C | Occupied
9/5/2021, 10:47:38 PM ecobee Bridge Ecobee 3 Sensors [Office] 22.2°C | Occupied
9/5/2021, 10:47:38 PM ecobee Bridge Ecobee 3 Sensors [Faber] 22.8°C | 48% | Occupied
9/5/2021, 10:47:38 PM ecobee Bridge Ecobee 3 Sensors Wait | 30 seconds
9/5/2021, 10:47:44 PM Hue Bridge Hue Philips hue: request 20: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:47:44 PM Hue Bridge Hue [ WARNING ] Philips hue: request 20: api error 101: link button not pressed
9/5/2021, 10:47:44 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s
9/5/2021, 10:47:59 PM Hue Bridge Hue Philips hue: request 21: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:47:59 PM Hue Bridge Hue [ WARNING ] Philips hue: request 21: api error 101: link button not pressed
9/5/2021, 10:47:59 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s
9/5/2021, 10:48:08 PM ecobee Bridge Ecobee 3 Sensors Update sensors
9/5/2021, 10:48:08 PM ecobee Bridge Ecobee 3 Sensors [Upstairs] 22.4°C | Occupied
9/5/2021, 10:48:08 PM ecobee Bridge Ecobee 3 Sensors [Office] 22.2°C | Occupied
9/5/2021, 10:48:08 PM ecobee Bridge Ecobee 3 Sensors [Faber] 22.8°C | 48% | Occupied
9/5/2021, 10:48:08 PM ecobee Bridge Ecobee 3 Sensors Wait | 30 seconds
9/5/2021, 10:48:14 PM Hue Bridge Hue Philips hue: request 22: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:48:14 PM Hue Bridge Hue [ WARNING ] Philips hue: request 22: api error 101: link button not pressed
9/5/2021, 10:48:14 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s
9/5/2021, 10:48:29 PM Hue Bridge Hue Philips hue: request 23: POST / {"devicetype":"homebridge-hue#hoobs"}
9/5/2021, 10:48:29 PM Hue Bridge Hue [ WARNING ] Philips hue: request 23: api error 101: link button not pressed
9/5/2021, 10:48:29 PM Hue Bridge Hue Philips hue: press link button on the bridge to create a user - retrying in 15s

```

**Additional context**

**Comment by EddieDSuza** [comment]
Run the link pairing sequence again and generate a new user key or if you are not using the plugin - uninstall it.

**Comment by Calorion** [comment]
I haven't tried to pair it yet because I installed it for future use. The question is, why is a stopped bridge allowing the plugins in it to do anything at all? I thought one of the benefits of Bridges was that you can stop them individually for troubleshooting purposes. If they are still active when off, that defeats that purpose.

**Comment by mkellsy** [comment]
The Hue plugin spawns a separate process for hub linking. The only way to stop this plugin is to link it to a hub. If you don't have a hub to link to yet, uninstall the plugin until you are ready to use it. After you uninstall it you will need to reboot the device to terminate this plugin.

**Comment by Calorion** [comment]
Seems like that violates the sandboxing security concept behind having bridges in the first place, but whatever.

**Comment by mkellsy** [comment]
The Hue plugin specifically detaches the child process from the parent. Not much I can do about it :(

**Comment by Calorion** [comment]
I'm not a programmer, so maybe I don't know what I'm talking about, but that sounds like insufficient sandboxing. If malicious programs (not saying Hue plugin is malicious) can spawn processes outside the bridge, then couldn't that compromise the security/stability of the device, or even perhaps the network?

**Comment by mkellsy** [comment]
I totally agree with you. However this is a lot bigger issue then this. Anyone who writes a plugin can do this. It is part of NodeJS.
https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options

Notice there is an option "detached". Specifically this
https://nodejs.org/api/child_process.html#child_process_options_detached

This allows plugin devs to write code that overruns the parent process. We, along with a very large community of developers have petitioned NodeJS to add some kind of security controls around child processes.

I have tried to solve this issue for a very long time. NodeJS just doesn't give the parent process control of detached processes. I have raised this issue with the plugin dev, but he brushed me off. It is this exact plugin that made me think about the potential malicious plugin problem that both HOOBS and Homebridge suffer from. To address this, we run our own NPM mirror for the plugins. This allows us to be a gatekeeper for these plugins and we can block ones that cause harm.

Currently we have blocked some plugins due to unintended bad code, plugins that do nothing and severely un-maintained code. We have blocked one plugin due to a phone home tracking. We reviewed blocking Hue due to the overrun issue, but the plugin is too popular and outside of the hub linking this plugin is solid.

Basically if I were to block this behavior, I would have to change runtimes. Essentially taking our plugin count from 3,000+ to 0. It's a necessary evil, and with proper monitoring we can keep this in check.
